packages/visual-editor/src/utils/resolveTranslatableString.tsx

```diff
@@ -3,9 +3,110 @@ import {
   RichText,
   TranslatableRichText,
   TranslatableString,
+  YextEntityField,
 } from "@yext/visual-editor";
 import React from "react";
-import { resolveEmbeddedFieldsInString } from "./resolveYextEntityField";
+import {
+  resolveEmbeddedFieldsRecursively,
+  resolveEmbeddedFieldsInString,
+  resolveYextEntityField,
+} from "./resolveYextEntityField";
+
+/**
+ * The primary function for resolving all component data. It handles entity
+ * fields, constant values, embedded fields, and translatable types.
+ *
+ * @param data The field configuration object from component props OR a direct translatable value.
+ * @param locale The current language locale (e.g., "en").
+ * @param document The entity document. If not provided, embedded fields will not be resolved.
+ * @returns The fully resolved data for the component.
+ */
+// --- Overload Signatures ---
+// 1. Handles TranslatableString directly or via a YextEntityField
+export function resolveComponentData(
```

Comment:
This should be the only thing exported now for users to use. The old ones should be internal to us now.